[v3.3.0] "Module not found: Error: Can't resolve 'fs'" with next.js

**Logs and screenshots**
<details>
<summary>log output</summary>
<pre>
&gt; $  DEBUG=code-coverage npm run start:e2e                                                          ⬡ 13.12.0 [±dependabot/npm_and_yarn/cypress/code-coverage-3.3.0 ●]

&gt; mui-next-ts@1.0.0 start:e2e /home/user/projects/mui-next-ts
&gt; start-test dev 3000 test:e2e

starting server using command "npm run dev"
and when url "[ 'http://localhost:3000' ]" is responding with HTTP status code 200
running tests using command "npm run test:e2e"

&gt; mui-next-ts@1.0.0 dev /home/user/projects/mui-next-ts
&gt; next dev

[ wait ]  starting the development server ...
[ info ]  waiting on http://localhost:3000 ...
&gt; Using external babel configuration
&gt; Location: "/home/user/projects/mui-next-ts/babel.config.js"
[ info ]  bundled successfully, waiting for typecheck results...
[ event ] build page: /
[ wait ]  compiling ...
[ event ] build page: /
[ info ]  bundled successfully, waiting for typecheck results...
[ event ] build page: /login
[ wait ]  compiling ...
[ event ] build page: /login
[ info ]  bundled successfully, waiting for typecheck results...

&gt; mui-next-ts@1.0.0 test:e2e /home/user/projects/mui-next-ts
&gt; cypress run

[ ready ] compiled successfully - ready on http://localhost:3000

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:    4.3.0                                                                              │
  │ Browser:    Electron 80 (headless)                                                             │
  │ Specs:      3 found (login.spec.ts, main.spec.ts, register.spec.ts)                            │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────

  Running:  login.spec.ts                                                                   (1 of 3)

Oops...we found an error preparing this test file:

  cypress/support/index.js

The error was:

Error: Webpack Compilation Error
./node_modules/@cypress/code-coverage/utils.js
Module not found: Error: Can't resolve 'fs' in '/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage'
resolve 'fs' in '/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage'
  Parsed request is a module
  using description file: /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      /home/user/projects/mui-next-ts/node_modules/@cypress/node_modules doesn't exist or is not a directory
      /home/user/projects/mui-next-ts/node_modules/node_modules doesn't exist or is not a directory
      /home/user/projects/node_modules doesn't exist or is not a directory
      /home/user/node_modules doesn't exist or is not a directory
      /home/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules
        using description file: /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
      looking for modules in /home/user/projects/mui-next-ts/node_modules
        using description file: /home/user/projects/mui-next-ts/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
          using description file: /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/package.json (relative path: ./node_modules/fs)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
          using description file: /home/user/projects/mui-next-ts/package.json (relative path: ./node_modules/fs)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/fs doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/fs.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.tsx doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/fs.tsx doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.js doesn't exist
              /home/user/projects/mui-next-ts/node_modules/fs.js doesn't exist
            as directory
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs doesn't exist
            as directory
              /home/user/projects/mui-next-ts/node_modules/fs doesn't exist
[/home/user/projects/mui-next-ts/node_modules/@cypress/node_modules]
[/home/user/projects/mui-next-ts/node_modules/node_modules]
[/home/user/projects/node_modules]
[/home/user/node_modules]
[/home/node_modules]
[/node_modules]
[/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs]
[/home/user/projects/mui-next-ts/node_modules/fs]
[/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.ts]
[/home/user/projects/mui-next-ts/node_modules/fs.ts]
[/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.tsx]
[/home/user/projects/mui-next-ts/node_modules/fs.tsx]
[/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.js]
[/home/user/projects/mui-next-ts/node_modules/fs.js]
 @ ./node_modules/@cypress/code-coverage/utils.js 3:52-65
 @ ./node_modules/@cypress/code-coverage/support.js
 @ ./cypress/support/index.js
 @ multi ./cypress/support/index.js

This occurred while Cypress was compiling and bundling your test code. This is usually caused by:

- A missing file or dependency
- A syntax error in the file or one of its dependencies

Fix the error in your code and re-run your tests.

  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        0                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     0 seconds                                                                        │
  │ Spec Ran:     login.spec.ts                                                                    │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Video)

  -  Started processing:  Compressing to 32 CRF
  -  Finished processing: /home/user/projects/mui-next-ts/cypress/    (0 seconds)
                          videos/login.spec.ts.mp4


────────────────────────────────────────────────────────────────────────────────────────────────────

  Running:  main.spec.ts                                                                    (2 of 3)

Oops...we found an error preparing this test file:

  cypress/support/index.js

The error was:

Error: Webpack Compilation Error
./node_modules/@cypress/code-coverage/utils.js
Module not found: Error: Can't resolve 'fs' in '/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage'
resolve 'fs' in '/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage'
  Parsed request is a module
  using description file: /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      /home/user/projects/mui-next-ts/node_modules/@cypress/node_modules doesn't exist or is not a directory
      /home/user/projects/mui-next-ts/node_modules/node_modules doesn't exist or is not a directory
      /home/user/projects/node_modules doesn't exist or is not a directory
      /home/user/node_modules doesn't exist or is not a directory
      /home/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules
        using description file: /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
      looking for modules in /home/user/projects/mui-next-ts/node_modules
        using description file: /home/user/projects/mui-next-ts/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
          using description file: /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/package.json (relative path: ./node_modules/fs)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
          using description file: /home/user/projects/mui-next-ts/package.json (relative path: ./node_modules/fs)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/fs doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/fs.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.tsx doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/fs.tsx doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.js doesn't exist
              /home/user/projects/mui-next-ts/node_modules/fs.js doesn't exist
            as directory
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs doesn't exist
            as directory
              /home/user/projects/mui-next-ts/node_modules/fs doesn't exist
[/home/user/projects/mui-next-ts/node_modules/@cypress/node_modules]
[/home/user/projects/mui-next-ts/node_modules/node_modules]
[/home/user/projects/node_modules]
[/home/user/node_modules]
[/home/node_modules]
[/node_modules]
[/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs]
[/home/user/projects/mui-next-ts/node_modules/fs]
[/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.ts]
[/home/user/projects/mui-next-ts/node_modules/fs.ts]
[/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.tsx]
[/home/user/projects/mui-next-ts/node_modules/fs.tsx]
[/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.js]
[/home/user/projects/mui-next-ts/node_modules/fs.js]
 @ ./node_modules/@cypress/code-coverage/utils.js 3:52-65
 @ ./node_modules/@cypress/code-coverage/support.js
 @ ./cypress/support/index.js
 @ multi ./cypress/support/index.js

This occurred while Cypress was compiling and bundling your test code. This is usually caused by:

- A missing file or dependency
- A syntax error in the file or one of its dependencies

Fix the error in your code and re-run your tests.

  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        0                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     0 seconds                                                                        │
  │ Spec Ran:     main.spec.ts                                                                     │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Video)

  -  Started processing:  Compressing to 32 CRF
  -  Finished processing: /home/user/projects/mui-next-ts/cypress/    (0 seconds)
                          videos/main.spec.ts.mp4


────────────────────────────────────────────────────────────────────────────────────────────────────

  Running:  register.spec.ts                                                                (3 of 3)

Oops...we found an error preparing this test file:

  cypress/support/index.js

The error was:

Error: Webpack Compilation Error
./node_modules/@cypress/code-coverage/utils.js
Module not found: Error: Can't resolve 'fs' in '/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage'
resolve 'fs' in '/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage'
  Parsed request is a module
  using description file: /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      /home/user/projects/mui-next-ts/node_modules/@cypress/node_modules doesn't exist or is not a directory
      /home/user/projects/mui-next-ts/node_modules/node_modules doesn't exist or is not a directory
      /home/user/projects/node_modules doesn't exist or is not a directory
      /home/user/node_modules doesn't exist or is not a directory
      /home/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules
        using description file: /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
      looking for modules in /home/user/projects/mui-next-ts/node_modules
        using description file: /home/user/projects/mui-next-ts/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
          using description file: /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/package.json (relative path: ./node_modules/fs)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
          using description file: /home/user/projects/mui-next-ts/package.json (relative path: ./node_modules/fs)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/fs doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/fs.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.tsx doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/fs.tsx doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.js doesn't exist
              /home/user/projects/mui-next-ts/node_modules/fs.js doesn't exist
            as directory
              /home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs doesn't exist
            as directory
              /home/user/projects/mui-next-ts/node_modules/fs doesn't exist
[/home/user/projects/mui-next-ts/node_modules/@cypress/node_modules]
[/home/user/projects/mui-next-ts/node_modules/node_modules]
[/home/user/projects/node_modules]
[/home/user/node_modules]
[/home/node_modules]
[/node_modules]
[/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs]
[/home/user/projects/mui-next-ts/node_modules/fs]
[/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.ts]
[/home/user/projects/mui-next-ts/node_modules/fs.ts]
[/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.tsx]
[/home/user/projects/mui-next-ts/node_modules/fs.tsx]
[/home/user/projects/mui-next-ts/node_modules/@cypress/code-coverage/node_modules/fs.js]
[/home/user/projects/mui-next-ts/node_modules/fs.js]
 @ ./node_modules/@cypress/code-coverage/utils.js 3:52-65
 @ ./node_modules/@cypress/code-coverage/support.js
 @ ./cypress/support/index.js
 @ multi ./cypress/support/index.js

This occurred while Cypress was compiling and bundling your test code. This is usually caused by:

- A missing file or dependency
- A syntax error in the file or one of its dependencies

Fix the error in your code and re-run your tests.

  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        0                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     0 seconds                                                                        │
  │ Spec Ran:     register.spec.ts                                                                 │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Video)

  -  Started processing:  Compressing to 32 CRF
  -  Finished processing: /home/user/projects/mui-next-ts/cypress/    (0 seconds)
                          videos/register.spec.ts.mp4


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✖  login.spec.ts                              0ms        -        -        1        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  main.spec.ts                               0ms        -        -        1        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  register.spec.ts                           0ms        -        -        1        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖  3 of 3 failed (100%)                       0ms        -        -        3        -        -

npm ERR! code ELIFECYCLE
npm ERR! errno 3
npm ERR! mui-next-ts@1.0.0 test:e2e: `cypress run`
npm ERR! Exit status 3
npm ERR!
npm ERR! Failed at the mui-next-ts@1.0.0 test:e2e script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jsuncin/.npm/_logs/2020-04-19T17_01_15_539Z-debug.log
Error: Command failed with exit code 3: npm run test:e2e
    at makeError (/home/user/projects/mui-next-ts/node_modules/execa/lib/error.js:56:11)
    at handlePromise (/home/user/projects/mui-next-ts/node_modules/execa/index.js:114:26)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  command: 'npm run test:e2e',
  exitCode: 3,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mui-next-ts@1.0.0 start:e2e: `start-test dev 3000 test:e2e`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the mui-next-ts@1.0.0 start:e2e script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2020-04-19T17_01_15_591Z-debug.log
</pre>
</details>

[CI output](https://github.com/leosuncin/mui-next-ts/pull/169/checks?check_run_id=599316319)

**Versions**

- What is this plugin's version?
3.3.0
- What is Cypress version?
4.3.0
- What is your operating system?
Manjaro Linux
- What is the shell?
ZSH
- What is the Node version?
13.12.0
- What is the NPM version?
6.14.4
- Is there `.nyc_output` folder? Is there `.nyc_output/out.json` file. Is it empty? Can you paste at least part of it so we can see the keys and file paths?
Is empty
- Do you have any custom NYC settings in `package.json` (`nyc` object) or in other [NYC config files](https://github.com/istanbuljs/nyc#configuration-files)
```json
...
  "nyc": {
    "extends": "@istanbuljs/nyc-config-typescript",
    "all": true,
    "report-dir": "coverage/e2e",
    "exclude": [
      "pages/api/__coverage__.ts"
    ]
  }
...
```
- Do you run Cypress tests in a Docker container?
No, except from CI

**Describe the bug**
Next.js strip out `fs` module from dependency tree to avoid to be included in client bundle. In the last version (dfaed9b) this plugin require `fs` to work, so is incompatible with next.js.

**Link to the repo**
https://github.com/leosuncin/mui-next-ts/tree/dependabot/npm_and_yarn/cypress/code-coverage-3.3.0

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -17,8 +17,16 @@ jobs:
       - checkout
       - node/with-cache:
           steps:
-            - run: CYPRESS_INSTALL_BINARY=0 npm ci
-      - run: npm run format:check
+            - run:
+                name: Install dependencies 📦
+                # installs NPM dependencies but skips Cypress
+                command: CYPRESS_INSTALL_BINARY=0 npm ci
+      - run:
+          name: Code style check 🧹
+          command: npm run format:check
+      - run:
+          name: Types lint 🧹
+          command: npm run types
 
   publish:
     description: Publishes the new version of the plugin to NPM

--- a/cypress/integration/filtering.js
+++ b/cypress/integration/filtering.js
@@ -1,4 +1,4 @@
-const { filterSpecsFromCoverage } = require('../../utils')
+const { filterSpecsFromCoverage } = require('../../support-utils')
 
 describe('minimatch', () => {
   it('string matches', () => {

--- a/cypress/integration/spec.js
+++ b/cypress/integration/spec.js
@@ -3,7 +3,7 @@
 /// <reference types="Cypress" />
 
 import { add } from '../unit'
-const { fixSourcePathes } = require('../../utils')
+const { fixSourcePaths } = require('../../task-utils')
 
 context('Page test', () => {
   beforeEach(() => {
@@ -54,7 +54,7 @@ context('Unit tests', () => {
       }
     }
 
-    fixSourcePathes(coverage)
+    fixSourcePaths(coverage)
 
     expect(coverage).to.deep.eq({
       '/absolute/src/component.vue': {

--- a/cypress/integration/spec.js
+++ b/cypress/integration/spec.js
@@ -3,7 +3,7 @@
 /// <reference types="Cypress" />
 
 import { add } from '../unit'
-const { fixSourcePaths } = require('../../task-utils')
+const { fixSourcePaths } = require('../../support-utils')
 
 context('Page test', () => {
   beforeEach(() => {

--- a/package.json
+++ b/package.json
@@ -15,7 +15,8 @@
     "format": "prettier --write '*.js'",
     "format:check": "prettier --check '*.js'",
     "check:markdown": "find *.md -exec npx markdown-link-check {} \\;",
-    "effective:config": "circleci config process .circleci/config.yml | sed /^#/d"
+    "effective:config": "circleci config process .circleci/config.yml | sed /^#/d",
+    "types": "tsc --noEmit --allowJs *.js cypress/integration/*.js"
   },
   "peerDependencies": {
     "cypress": "*"

--- a/support-utils.js
+++ b/support-utils.js
@@ -1,3 +1,4 @@
+// @ts-check
 // helper functions that are safe to use in the browser
 // from support.js file - no file system access
 
@@ -40,6 +41,24 @@ const filterSpecsFromCoverage = (totalCoverage, config = Cypress.config) => {
 
   const coverage = Cypress._.omitBy(totalCoverage, isTestFileFilter)
   return coverage
+}
+
+/**
+ * Replace source-map's path by the corresponding absolute file path
+ * (coverage report wouldn't work with source-map path being relative
+ * or containing Webpack loaders and query parameters)
+ */
+function fixSourcePaths(coverage) {
+  Object.values(coverage).forEach(file => {
+    const { path: absolutePath, inputSourceMap } = file
+    const fileName = /([^\/\\]+)$/.exec(absolutePath)[1]
+    if (!inputSourceMap || !fileName) return
+
+    if (inputSourceMap.sourceRoot) inputSourceMap.sourceRoot = ''
+    inputSourceMap.sources = inputSourceMap.sources.map(source =>
+      source.includes(fileName) ? absolutePath : source
+    )
+  })
 }
 
 module.exports = {

--- a/support-utils.js
+++ b/support-utils.js
@@ -1,0 +1,48 @@
+// helper functions that are safe to use in the browser
+// from support.js file - no file system access
+
+/**
+ * remove coverage for the spec files themselves,
+ * only keep "external" application source file coverage
+ */
+const filterSpecsFromCoverage = (totalCoverage, config = Cypress.config) => {
+  const integrationFolder = config('integrationFolder')
+  // @ts-ignore
+  const testFilePattern = config('testFiles')
+
+  // test files chould be:
+  //  wild card string "**/*.*" (default)
+  //  wild card string "**/*spec.js"
+  //  list of wild card strings or names ["**/*spec.js", "spec-one.js"]
+  const testFilePatterns = Array.isArray(testFilePattern)
+    ? testFilePattern
+    : [testFilePattern]
+
+  const isUsingDefaultTestPattern = testFilePattern === '**/*.*'
+
+  const isTestFile = filename => {
+    const matchedPattern = testFilePatterns.some(specPattern =>
+      Cypress.minimatch(filename, specPattern)
+    )
+    const matchedEndOfPath = testFilePatterns.some(specPattern =>
+      filename.endsWith(specPattern)
+    )
+    return matchedPattern || matchedEndOfPath
+  }
+
+  const isInIntegrationFolder = filename =>
+    filename.startsWith(integrationFolder)
+
+  const isA = (fileCoverge, filename) => isInIntegrationFolder(filename)
+  const isB = (fileCoverge, filename) => isTestFile(filename)
+
+  const isTestFileFilter = isUsingDefaultTestPattern ? isA : isB
+
+  const coverage = Cypress._.omitBy(totalCoverage, isTestFileFilter)
+  return coverage
+}
+
+module.exports = {
+  fixSourcePaths,
+  filterSpecsFromCoverage
+}

--- a/support.js
+++ b/support.js
@@ -1,7 +1,7 @@
 /// <reference types="cypress" />
 // @ts-check
 
-const { filterSpecsFromCoverage } = require('./utils')
+const { filterSpecsFromCoverage } = require('./support-utils')
 
 /**
  * Sends collected code coverage object to the backend code

--- a/support.js
+++ b/support.js
@@ -158,7 +158,7 @@ const registerHooks = () => {
         failOnStatusCode: false
       })
         .then(r => {
-          return Cypress._.get(r, 'body.coverage', null), { log: false }
+          return Cypress._.get(r, 'body.coverage', null)
         })
         .then(coverage => {
           if (!coverage) {

--- a/support.js
+++ b/support.js
@@ -35,6 +35,7 @@ const logMessage = s => {
 const filterSupportFilesFromCoverage = totalCoverage => {
   const integrationFolder = Cypress.config('integrationFolder')
   const supportFile = Cypress.config('supportFile')
+  // @ts-ignore
   const supportFolder = Cypress.config('supportFolder')
 
   const isSupportFile = filename => filename === supportFile
@@ -61,12 +62,14 @@ const registerHooks = () => {
 
   const hasE2ECoverage = () => Boolean(windowCoverageObjects.length)
 
+  // @ts-ignore
   const hasUnitTestCoverage = () => Boolean(window.__coverage__)
 
   before(() => {
     // we need to reset the coverage when running
     // in the interactive mode, otherwise the counters will
     // keep increasing every time we rerun the tests
+    // @ts-ignore
     cy.task('resetCoverage', { isInteractive: Cypress.config('isInteractive') })
   })
 
@@ -133,7 +136,9 @@ const registerHooks = () => {
 
     // there might be server-side code coverage information
     // we should grab it once after all tests finish
+    // @ts-ignore
     const baseUrl = Cypress.config('baseUrl') || cy.state('window').origin
+    // @ts-ignore
     const runningEndToEndTests = baseUrl !== Cypress.config('proxyUrl')
     const specType = Cypress._.get(Cypress.spec, 'specType', 'integration')
     const isIntegrationSpec = specType === 'integration'
@@ -152,7 +157,9 @@ const registerHooks = () => {
         log: false,
         failOnStatusCode: false
       })
-        .then(r => Cypress._.get(r, 'body.coverage', null), { log: false })
+        .then(r => {
+          return Cypress._.get(r, 'body.coverage', null), { log: false }
+        })
         .then(coverage => {
           if (!coverage) {
             // we did not get code coverage - this is the
@@ -171,6 +178,7 @@ const registerHooks = () => {
     // then we will have unit test coverage
     // NOTE: spec iframe is NOT reset between the tests, so we can grab
     // the coverage information only once after all tests have finished
+    // @ts-ignore
     const unitTestCoverage = window.__coverage__
     if (unitTestCoverage) {
       sendCoverage(unitTestCoverage, 'unit')

--- a/task-utils.js
+++ b/task-utils.js
@@ -194,28 +194,9 @@ function tryFindingLocalFiles(nycFilename) {
   }
 }
 
-/**
- * Replace source-map's path by the corresponding absolute file path
- * (coverage report wouldn't work with source-map path being relative
- * or containing Webpack loaders and query parameters)
- */
-function fixSourcePaths(coverage) {
-  Object.values(coverage).forEach(file => {
-    const { path: absolutePath, inputSourceMap } = file
-    const fileName = /([^\/\\]+)$/.exec(absolutePath)[1]
-    if (!inputSourceMap || !fileName) return
-
-    if (inputSourceMap.sourceRoot) inputSourceMap.sourceRoot = ''
-    inputSourceMap.sources = inputSourceMap.sources.map(source =>
-      source.includes(fileName) ? absolutePath : source
-    )
-  })
-}
-
 module.exports = {
   showNycInfo,
   resolveRelativePaths,
   checkAllPathsNotFound,
-  tryFindingLocalFiles,
-  fixSourcePaths
+  tryFindingLocalFiles
 }

--- a/task-utils.js
+++ b/task-utils.js
@@ -1,3 +1,6 @@
+// helper functions to use from "task.js" plugins code
+// that need access to the file system
+
 // @ts-check
 /// <reference types="cypress" />
 const { readFileSync, writeFileSync, existsSync } = require('fs')
@@ -24,65 +27,6 @@ function checkAllPathsNotFound(nycFilename) {
     allFilesAreMissing
   )
   return allFilesAreMissing
-}
-
-/**
- * remove coverage for the spec files themselves,
- * only keep "external" application source file coverage
- */
-const filterSpecsFromCoverage = (totalCoverage, config = Cypress.config) => {
-  const integrationFolder = config('integrationFolder')
-  // @ts-ignore
-  const testFilePattern = config('testFiles')
-
-  // test files chould be:
-  //  wild card string "**/*.*" (default)
-  //  wild card string "**/*spec.js"
-  //  list of wild card strings or names ["**/*spec.js", "spec-one.js"]
-  const testFilePatterns = Array.isArray(testFilePattern)
-    ? testFilePattern
-    : [testFilePattern]
-
-  const isUsingDefaultTestPattern = testFilePattern === '**/*.*'
-
-  const isTestFile = filename => {
-    const matchedPattern = testFilePatterns.some(specPattern =>
-      Cypress.minimatch(filename, specPattern)
-    )
-    const matchedEndOfPath = testFilePatterns.some(specPattern =>
-      filename.endsWith(specPattern)
-    )
-    return matchedPattern || matchedEndOfPath
-  }
-
-  const isInIntegrationFolder = filename =>
-    filename.startsWith(integrationFolder)
-
-  const isA = (fileCoverge, filename) => isInIntegrationFolder(filename)
-  const isB = (fileCoverge, filename) => isTestFile(filename)
-
-  const isTestFileFilter = isUsingDefaultTestPattern ? isA : isB
-
-  const coverage = Cypress._.omitBy(totalCoverage, isTestFileFilter)
-  return coverage
-}
-
-/**
- * Replace source-map's path by the corresponding absolute file path
- * (coverage report wouldn't work with source-map path being relative
- * or containing Webpack loaders and query parameters)
- */
-function fixSourcePathes(coverage) {
-  Object.values(coverage).forEach(file => {
-    const { path: absolutePath, inputSourceMap } = file
-    const fileName = /([^\/\\]+)$/.exec(absolutePath)[1]
-    if (!inputSourceMap || !fileName) return
-
-    if (inputSourceMap.sourceRoot) inputSourceMap.sourceRoot = ''
-    inputSourceMap.sources = inputSourceMap.sources.map(source =>
-      source.includes(fileName) ? absolutePath : source
-    )
-  })
 }
 
 /**
@@ -250,11 +194,28 @@ function tryFindingLocalFiles(nycFilename) {
   }
 }
 
+/**
+ * Replace source-map's path by the corresponding absolute file path
+ * (coverage report wouldn't work with source-map path being relative
+ * or containing Webpack loaders and query parameters)
+ */
+function fixSourcePaths(coverage) {
+  Object.values(coverage).forEach(file => {
+    const { path: absolutePath, inputSourceMap } = file
+    const fileName = /([^\/\\]+)$/.exec(absolutePath)[1]
+    if (!inputSourceMap || !fileName) return
+
+    if (inputSourceMap.sourceRoot) inputSourceMap.sourceRoot = ''
+    inputSourceMap.sources = inputSourceMap.sources.map(source =>
+      source.includes(fileName) ? absolutePath : source
+    )
+  })
+}
+
 module.exports = {
-  fixSourcePathes,
-  filterSpecsFromCoverage,
   showNycInfo,
   resolveRelativePaths,
   checkAllPathsNotFound,
-  tryFindingLocalFiles
+  tryFindingLocalFiles,
+  fixSourcePaths
 }

--- a/task.js
+++ b/task.js
@@ -4,12 +4,12 @@ const { join, resolve } = require('path')
 const { existsSync, mkdirSync, readFileSync, writeFileSync } = require('fs')
 const execa = require('execa')
 const {
-  fixSourcePaths,
   showNycInfo,
   resolveRelativePaths,
   checkAllPathsNotFound,
   tryFindingLocalFiles
 } = require('./task-utils')
+const { fixSourcePaths } = require('./support-utils')
 const NYC = require('nyc')
 
 const debug = require('debug')('code-coverage')

--- a/task.js
+++ b/task.js
@@ -4,12 +4,12 @@ const { join, resolve } = require('path')
 const { existsSync, mkdirSync, readFileSync, writeFileSync } = require('fs')
 const execa = require('execa')
 const {
-  fixSourcePathes,
+  fixSourcePaths,
   showNycInfo,
   resolveRelativePaths,
   checkAllPathsNotFound,
   tryFindingLocalFiles
-} = require('./utils')
+} = require('./task-utils')
 const NYC = require('nyc')
 
 const debug = require('debug')('code-coverage')

--- a/task.js
+++ b/task.js
@@ -79,7 +79,7 @@ const tasks = {
     const coverage = JSON.parse(sentCoverage)
     debug('parsed sent coverage')
 
-    fixSourcePathes(coverage)
+    fixSourcePaths(coverage)
     const previous = existsSync(nycFilename)
       ? JSON.parse(readFileSync(nycFilename, 'utf8'))
       : istanbul.createCoverageMap({})

